docs/user_guide/text_parser.md

```diff
@@ -0,0 +1,178 @@
+# text_paser
+
+The module `text_parser` provides rule based text parser that is closely modeled
+after the Ansible playbook language. This parser will iterate of the rules and
+parse the output of structured ASCII text into a JSON data structure that can be
+added to the inventory host facts.
+
+## Playbook
+
+```yaml
+
+---
+# The following task runs on localhost.
+
+- hosts: localhost
+
+  tasks:
+  - name: Genereta interface facts as JSON
+    text_parser:
+      file: "parsers/ios/show_interfaces.yaml"
+      content: "{{ lookup('file', 'output/ios/show_interfaces.txt') }}"
+
+  - name: Generate system facts as JSON
+    text_parser:
+      file: "parsers/ios/show_version.yaml"
+      content: "{{ lookup('file', 'output/ios/show_version.txt') }}"
+
+
+# The following task runs against network device
+
+- hosts: ios
+
+  tasks:
+  - name: Collect interface information from device
+    ios_command:
+      commands: "show interfaces"
+    register: ios_interface_output
+
+  - name: Generate interface facts as JSON
+    text_parser:
+      file: "parsers/ios/show_interfaces.yaml"
+      content: ios_interface_output['stdout'][0]
+
+```
+
+## Parser
+The `file` parameter for `text_parser` contains rules to parse text.
+The rules in parser file uses directives written closely to Ansible language.
+
+Directives documentation is available [Here](https://github.com/ansible-network/network-engine/blob/devel/docs/directives/parser_directives.md).
```

Comment:
suggestion: Directives documentation is available in [parser_directives.md](https://github.com/ansible-network/network-engine/blob/devel/docs/directives/parser_directives.md).